From 269c8922efd998023f8f94bb23ba94265332be50 Mon Sep 17 00:00:00 2001
From: Ming Jiang <mjiang@qti.qualcomm.com>
Date: Wed, 10 Sep 2025 15:54:18 +0800
Subject: [PATCH] dts configure for genoa wlan card

wlan dts inserted in imx8-ss-hsio.dtsi for imx8qxp

Upstream-Status: Pending
Signed-off-by: Ming Jiang <mjiang@qti.qualcomm.com>
---
 .../boot/dts/freescale/imx8-ss-hsio.dtsi      | 38 +++++++++++++++++++
 1 file changed, 38 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/imx8-ss-hsio.dtsi b/arch/arm64/boot/dts/freescale/imx8-ss-hsio.dtsi
index 469de8b536b5..acc42f8b6c3e 100644
--- a/arch/arm64/boot/dts/freescale/imx8-ss-hsio.dtsi
+++ b/arch/arm64/boot/dts/freescale/imx8-ss-hsio.dtsi
@@ -131,6 +131,44 @@ pcieb_crr3_lpcg: clock-controller@5f0d0000 {
 		power-domains = <&pd IMX_SC_R_PCIE_B>;
 	};
 
+	mhi: qcom,mhi {
+		compatible = "qcom,mhi";
+		status = "okay";
+	};
+
+	ipc_router: qcom,ipc_router_external_modem_xprt2 {
+		compatible = "qcom,ipc_router_mhi_xprt";
+		qcom,ch-name = "IPCRTR";
+		qcom,out-chan-id = <16>;
+		qcom,in-chan-id = <17>;
+		qcom,xprt-remote = "external-wlan";
+		qcom,xprt-linkid = <1>;
+		qcom,xprt-version = <3>;
+		status = "okay";
+	};
+
+	diag_mhi: qcom,diag-mhi {
+		compatible = "qcom,diag-mhi";
+		status = "okay";
+	};
+
+	cnss2: qcom,cnss-qca-converged {
+		compatible = "qcom,cnss-qca-converged";
+		qcom,converged-dt;
+		qcom,bus-type=<0>;
+
+		status = "okay";
+
+		/* Support Genoa Device */
+		chip_cfg@1 {
+			supported-ids = <0x1102>;
+		};
+
+		/* Support Hasting Prime Device */
+		chip_cfg@3 {
+			supported-ids = <0x1103>;
+		};
+	};
 	misc_crr5_lpcg: clock-controller@5f0f0000 {
 		compatible = "fsl,imx8qxp-lpcg";
 		reg = <0x5f0f0000 0x10000>;

-- 
2.34.1

